File: test/fixtures/sample-sdk-python/src/workos/sso.py
from typing import Optional, Protocol
from workos.types.sso.connection import Connection
from workos.types.sso.profile import Profile


class SSOModule(Protocol):
    def get_connection(self, connection_id: str) -> Connection: ...

    def list_connections(
        self,
        *,
        connection_type: Optional[str] = None,
        organization_id: Optional[str] = None,
        limit: int = 10,
    ) -> list: ...

    def get_profile(self, access_token: str) -> Profile: ...

File: test/fixtures/sample-sdk-php/lib/SSO.php
<?php

namespace WorkOS;

class SSO
{
    /**
     * Get an authorization URL.
     *
     * @param string $domain
     * @param string $redirectUri
     * @param null|array $state
     * @param null|string $provider
     * @param null|string $connectionId
     * @param null|string $organizationId
     *
     * @return string
     */
    public function getAuthorizationUrl($domain, $redirectUri, $state = null, $provider = null, $connectionId = null, $organizationId = null)
    {
        // implementation
    }

    /**
     * Get a profile and token.
     *
     * @param string $code
     *
     * @return \WorkOS\Resource\Profile
     */
    public function getProfileAndToken($code)
    {
        // implementation
    }

    /**
     * Get a connection.
     *
     * @param string $connectionId
     *
     * @return \WorkOS\Resource\Connection
     */
    public function getConnection($connectionId)
    {
        // implementation
    }

    /**
     * List connections.
     *
     * @param null|string $connectionType
     * @param null|string $domain
     * @param null|string $organizationId
     * @param int $limit
     * @param null|string $before
     * @param null|string $after
     *
     * @return array
     */
    public function listConnections($connectionType = null, $domain = null, $organizationId = null, $limit = 10, $before = null, $after = null)
    {
        // implementation
    }
}
